### ISSUE TITLE
Not all tests pass

### ISSUE THREAD

**Issue by Taitava** [Taitava/cb-PatternCompare#1]
```
cbUnit test report 09 Mar 2021 20:50:26:

..\GeneralTextCompare\tests\test_GeneralTextCompare.cb:

  Failures:
    test_Wild():
      GeneralTextCompare(*issip*ss*, mississipissippi) should match. ()
    test_Wild():
      GeneralTextCompare(a*b, a*abab) should match. ()
    test_Wild():
      GeneralTextCompare(a*, a*r) should match. ()
    test_Wild():
      GeneralTextCompare(*aabbaa*a*, aaabbaabbaab) should match. ()

  173 total assertions.
  169 succeeded assertions.
  4 failed assertions.
```

**Comment by Taitava** [comment]
I double checked the `GeneralTextCompare()` function that I've ported to CoolBasic and compared it to the original algorithm. No mistakes had happened during the porting, so it seems that the original algorithm is defective. Well, I did pick an old version of the algorithm, but I didn't expect that it would have this kind of errors.

The test cases are from the newer version of the algorithm, because I didn't find tests for the old version. Maybe good that it went this way.

I also tested to run the algorithm's old version's C++ code in an online C++ interpreter with the tests. I was able to get it compiled and run, but the tests failed there too, so it's also another proof that the old version of the algorithm really is defective. I did the same run with the newer version of the C++ algorithm, and it passed the tests.

I think I'll just have to port the newer version of the algorithm - although I don't like it so much. It lacks the case sensitivity parameter and alternative terminator parameter. The latter is not so important (AFAIK), but the case sensitivity toggle is mandatory in my opinion. I don't even know if the newer algorithm is case-sensitive or not, so if I want to implement the toggle, I'll first need to find out which way it currently works.
